modepy/modes.py

```diff
@@ -79,22 +74,52 @@
 .. currentmodule:: modepy.modes
 
 .. autofunction:: pkdo_2d(order, rs)
-
 .. autofunction:: grad_pkdo_2d(order, rs)
-
 .. autofunction:: pkdo_3d(order, rst)
-
 .. autofunction:: grad_pkdo_3d(order, rst)
 
 Monomials
 ---------
 
 .. autofunction:: monomial(order, rst)
-
 .. autofunction:: grad_monomial(order, rst)
+
+Symbolic Basis Functions
+------------------------
+.. autofunction:: symbolicize_basis(order, rst)
 """
 
 
+# {{{ helpers for symbolic evaluation
+
+def _cse(expr, prefix):
+    if "pymbolic" in sys.modules:
```

Comment:
Why not a `try.. except ImportError`?

Reply:
Because that'll have an import as a side effect.